docs/rules/pair-to-have-been-called-assertions.md

```diff
@@ -0,0 +1,142 @@
+pass even if the mock was called more times than expected, potentially masking
+bugs.
+
+This rule requires that whenever you use these matchers with arguments, you must
+also use the corresponding `toHaveBeenCalledTimes()` or `toBeCalledTimes()`
+matcher for the same mock function to ensure an exact call count.
+
+### Benefits
+
+- **Prevents false positives**: Ensures tests fail when a mock is called more
+  times than expected
+- **Makes test intentions explicit**: Clearly documents how many times a
+  function should be called
+- **Improves test reliability**: Catches unexpected behavior where functions are
+  called multiple times
+- **Follows testing best practices**: Encourages complete and precise assertions
+
+## Examples
+
+### Incorrect
+
+```js
+expect(mockFn).toHaveBeenCalledWith('arg');
+
+expect(mockFn).toBeCalledWith('arg');
+
+// Multiple assertions without call count check
+expect(mockFn).toHaveBeenCalledWith('arg1');
+expect(mockFn).toHaveBeenCalledWith('arg2');
+
+// Using toHaveBeenNthCalledWith without call count
+expect(mockFn).toHaveBeenNthCalledWith(1, 'first');
+
+// Using toHaveBeenLastCalledWith without call count
+expect(mockFn).toHaveBeenLastCalledWith('last');
+
+// Using toBeNthCalledWith without call count
+expect(mockFn).toBeNthCalledWith(2, 'second');
+
+// Using toBeLastCalledWith without call count
+expect(mockFn).toBeLastCalledWith('last');
+```
+
+### Correct
+
+```js
+expect(mockFn).toHaveBeenCalledTimes(1);
+expect(mockFn).toHaveBeenCalledWith('arg');
+
+expect(mockFn).toBeCalledTimes(1);
+expect(mockFn).toBeCalledWith('arg');
+
+// Multiple mocks, each with call count
+expect(mockFn1).toHaveBeenCalledTimes(1);
+expect(mockFn1).toHaveBeenCalledWith('arg1');
+expect(mockFn2).toHaveBeenCalledTimes(1);
+expect(mockFn2).toHaveBeenCalledWith('arg2');
+
+// Using toHaveBeenNthCalledWith with call count
+expect(mockFn).toHaveBeenCalledTimes(2);
+expect(mockFn).toHaveBeenNthCalledWith(1, 'first');
+
+// Using toHaveBeenLastCalledWith with call count
+expect(mockFn).toHaveBeenCalledTimes(3);
+expect(mockFn).toHaveBeenLastCalledWith('last');
+
+// Using toBeNthCalledWith with call count
+expect(mockFn).toBeCalledTimes(2);
+expect(mockFn).toBeNthCalledWith(2, 'second');
+
+// Using toBeLastCalledWith with call count
+expect(mockFn).toBeCalledTimes(1);
+expect(mockFn).toBeLastCalledWith('only');
+
+// Mixed matchers with call count
+expect(mockFn).toHaveBeenCalledTimes(3);
+expect(mockFn).toHaveBeenCalledWith('arg1');
+expect(mockFn).toHaveBeenNthCalledWith(2, 'arg2');
+expect(mockFn).toHaveBeenLastCalledWith('arg3');
+
+// Empty call (no arguments) doesn't require call count
+expect(mockFn).toHaveBeenCalledWith();
+
+// Using 'not' modifier doesn't require call count
+expect(mockFn).not.toHaveBeenCalledWith('arg');
+
+// Only checking call count is fine
+expect(mockFn).toHaveBeenCalledTimes(1);
+```
+
+## Additional Checks
+
+### Contradictory Assertions
```

Comment:
At this point I think this is out of scope for this rule - can we remove it for now?